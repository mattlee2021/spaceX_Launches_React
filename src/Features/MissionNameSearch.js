import UpdateSubmitHook from "../CustomHooks/updateAndSubmit";

/**
 * This function allows a user to search for SpaceX Launches by mission name. A user can type in any String
 * and any SpaceX mission that contains the inputted String will be returned. The clearSearch_state button
 * resets searchTerm_state to an empty String, and thus clears out the search box. To
 * reset the search, the user should click the submit button, after clicking the clear search button. 
 * Similarly, a user can manually clear out the search box and hit submit to reset the search. 
 */

export function MissionNameSearch(props) {

	const { onSubmit } = props;
	const { handleChange, clearSearch_state, this_state, handleSubmitSearch } = UpdateSubmitHook(onSubmit);

	return (
		<form onSubmit={handleSubmitSearch}>
			<div className="Filtering">
				<input
					className="Search_Mission_box"
					type="text"
					placeholder="Search For Any Past SpaceX Mission"
					value={this_state}
					onChange={handleChange}
				/>
				<button type="submit">Search</button>
				<button type="reset" onClick={clearSearch_state}>Clear Search</button>
			</div>
		</form>
	);



}