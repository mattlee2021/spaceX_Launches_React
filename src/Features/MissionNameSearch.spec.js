import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestRenderer from "react-test-renderer";
import MissionNameSearch from "./MissionNameSearch";

describe("MissionNameSearch", () => {
  it("Should match the snapshot", () => {
    const testRenderer = TestRenderer.create(<MissionNameSearch />);
    testRenderer.toJSON();
    expect(testRenderer).toMatchSnapshot();
  });

  it("Should call setMissionNameSearch with the entered input", () => {
    const mockSetMissionNameSearch = jest.fn();
    render(
      <MissionNameSearch setMissionNameSearch={mockSetMissionNameSearch} />
    );
    const inputBox = screen.getByRole("textbox");
    userEvent.type(inputBox, "Falcon");
    const submitButton = screen.getByRole("button", { name: "Search" });
    userEvent.click(submitButton);
    expect(mockSetMissionNameSearch).toHaveBeenCalledWith("Falcon");
  });

  it("Should clear input when clicking Clear Search", () => {
    render(<MissionNameSearch />);
    const inputBox = screen.getByRole("textbox");
    userEvent.type(inputBox, "Falcon");
    const clearSearchButton = screen.getByRole("button", {
      name: "Clear Search",
    });
    expect(inputBox).toHaveValue("Falcon");
    userEvent.click(clearSearchButton);
    expect(inputBox).toHaveValue("");
  });
});
