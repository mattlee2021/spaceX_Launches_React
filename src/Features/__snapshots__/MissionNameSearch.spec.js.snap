// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionNameSearch Should match the snapshot 1`] = `
<form>
  <input
    className="search__input"
    onChange={[Function]}
    placeholder="Search For Any Past SpaceX Mission"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Search
  </button>
  <button
    onClick={[Function]}
    type="reset"
  >
    Clear Search
  </button>
</form>
`;
